### PR TITLE
[CALCITE-2345] Running test with Fongo / real Mongo depending on maven profile

### PR DESCRIPTION
**Pull request by asereda-gs**
Updated Mongo test to run against [Fongo](https://github.com/fakemongo/fongo) during unit tests (surefire plugin) and connect to real mongo instance (running on `localhost`) during integration tests (failsafe plugin).  

Code will detect at runtime current maven profile (using `systemPropertyVariables`) and connect to correct instance.

### CODE REVIEW

**Comment by michaelmior** [comment]
It looks like the comment about `ENABLED` at the beginning `MongoAdapterTest.java` is now incorrect. The tests are not just included in the integration profile anymore.

**Comment by asereda-gs** [comment]
I have moved `ENABLED` flag to `MongoAdapterIT`.  There is still an edge case in the sense that `MongoDatabaseRule` will attempt to connect to the database even if `calcite.test.mongodb=false`. Is this an issue ?

**Comment by michaelmior** [comment]
If it's not too complicated to resolve, that would be good to fix. I don't think it's much of a practical concern though since it's always possible to exclude the mongodb project from testing which will have the same impact.

**Comment by asereda-gs** [comment]
This was addressed in the last (squashed) commit.

File: mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoDatabaseRule.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+   * Create an instance based on current maven profile (as defined by {@code -Pit}).
+   *
+   * @see MongoAssertions#integrationTestsProfile()
+   */
+  static MongoDatabaseRule create() {
+    final MongoClient client;
+    if (MongoAssertions.mongoIntegrationTestsEnabled()) {
+      // use to real client (connects to mongo)
+      client = new MongoClient("localhost");
+    } else {
```
Comment (michaelmior) [inline mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoDatabaseRule.java]: Are these tests still disabled if `calcite.test.mongodb` is false?

Comment (asereda-gs) [reply mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoDatabaseRule.java]: Yes.  Both have to be enabled `IT` and `calcite.test.mongodb`. See [mongoIntegrationTestsEnabled()](https://github.com/apache/calcite/pull/723/files#diff-22c56e58bcc144483d3b89ca12f380deR95)
```sql
integrationTestsProfile() && Util.getBooleanProperty("calcite.test.mongodb", true);
```

Comment (asereda-gs) [reply mongodb/src/test/java/org/apache/calcite/adapter/mongodb/MongoDatabaseRule.java]: I have simplified logic in `MongoAssertions` you now have self-explanatory methods: `useMongo()` and `useFongo()`

**Comment by michaelmior** [comment]
Thanks for the updates! Merged.

**Pull request closed without merging**